tests/datasource/data_connector/test_configured_asset_azure_data_connector.py

```diff
@@ -0,0 +1,1346 @@
+            data_connector_name="general_azure_data_connector",
+            data_asset_name="TestFiles",
+            batch_identifiers=IDDict(
+                {"name": "alex", "timestamp": "20200819", "price": "1300"}
+            ),
+        ),
+    ]
+    return expected
+
+
+@mock.patch(
+    "great_expectations.core.usage_statistics.usage_statistics.UsageStatisticsHandler.emit"
+)
+@mock.patch(
+    "great_expectations.datasource.data_connector.configured_asset_azure_data_connector.list_azure_keys"
+)
+@mock.patch(
+    "great_expectations.datasource.data_connector.configured_asset_azure_data_connector.BlobServiceClient"
+)
+def test_return_all_batch_definitions_unsorted_without_named_data_asset_name(
+    mock_azure_conn,
+    mock_list_keys,
+    mock_emit,
+    empty_data_context_stats_enabled,
+    expected_batch_definitions_unsorted,
+):
+    my_data_connector_yaml = yaml.load(
+        f"""
+           class_name: ConfiguredAssetAzureDataConnector
+           datasource_name: test_environment
+           execution_engine:
+               class_name: PandasExecutionEngine
+           container: my_container
+           name_starts_with: ""
+           assets:
+               TestFiles:
+           default_regex:
+               pattern: (.+)_(.+)_(.+)\\.csv
+               group_names:
+                   - name
+                   - timestamp
+                   - price
+       """,
+    )
+
+    mock_list_keys.return_value = [
+        "alex_20200809_1000.csv",
+        "eugene_20200809_1500.csv",
+        "james_20200811_1009.csv",
+        "abe_20200809_1040.csv",
+        "will_20200809_1002.csv",
+        "james_20200713_1567.csv",
+        "eugene_20201129_1900.csv",
+        "will_20200810_1001.csv",
+        "james_20200810_1003.csv",
+        "alex_20200819_1300.csv",
+    ]
+
+    my_data_connector: ConfiguredAssetAzureDataConnector = (
+        instantiate_class_from_config(
+            config=my_data_connector_yaml,
+            runtime_environment={
+                "name": "general_azure_data_connector",
+                "datasource_name": "test_environment",
+            },
+            config_defaults={
+                "module_name": "great_expectations.datasource.data_connector"
+            },
+        )
+    )
+
+    # In an actual production environment, Azure Blob Storage will automatically sort these blobs by path (alphabetic order).
+    # Source: https://docs.microsoft.com/en-us/rest/api/storageservices/List-Blobs?redirectedfrom=MSDN
+    #
+    # The expected behavior is that our `unsorted_batch_definition_list` will maintain the same order it parses through `list_azure_keys()` (hence "unsorted").
+    # When using an actual `BlobServiceClient` (and not a mock), the output of `list_azure_keys` would be pre-sorted by nature of how the system orders blobs.
+    # It is important to note that although this is a minor deviation, it is deemed to be immaterial as we still end up testing our desired behavior.
```

Comment:
We deviate from the S3 test here in one key way; `moto3` acts like a real S3 bucket and therefore sorts any blobs that it contains. As such, the output of methods that retrieve blobs will inherently provide a different ordering.

I believe this to be okay as we still ultimately test the behavior we're looking to evaluate.
* unsorted tests ensure that the input and output come out the same way (unsorted!)
* sorted tests ensure some type of ordering based on input criteria

I found it important to note this distinction since a real connection object would provide slightly different results. Alas, this is the price you pay with mocks 🐍